design/baremetal-operator/reboot-interface.md

```diff
@@ -138,6 +138,28 @@ The controller automatically removes all annotations with the
 
 - the Host is deprovisioned
 
+The API has been extended to permit either a hard or a soft reboot to
+take place, specified via the reboot annotation and a `mode` parameter. The API
+supports the following syntax:
+
+- `reboot.metal.io` -- immediate reboot via soft shutdown first,
+  followed by a hard shutdown if the soft shutdown fails.
+- `reboot.metal3.io: {'mode':'hard'}` -- immediate reboot via hard
+  shutdown, potentially allowing for high-availability use-cases.
+- `reboot.metal3.io/{key}` -- phased reboot, issued and managed by the
+  client registered with the `key`, via soft shutdown first, followed
+  by a hard reboot if the soft reboot fails.
+- `reboot.metal3.io/{key}: {'mode':'hard'}` -- phased reboot, issued
+  and managed by the client registered with the `key`, via hard
+  shutdown.
+
+Currently, the only supported annotation value is `{'mode':'hard'}`, if
+any other value is specified with the `reboot.metal3.io` annotation it will
+trigger the default behaviour to issue a soft reboot. It's important to
```

Comment:
This is mixing the default when we get an error with the description of the API.

The argument to the annotation isn't just a string or map, it's the representation of an object. The object has 1 member `mode` that can have 2 values of an enum, `hard` or `soft`. If we can't interpret the data given to us as fitting into that description, the default for `mode` is `soft`.

Reply:
Thanks, I've addressed this in the latest commit. WDYT?